### PR TITLE
dosomething_fact module

### LINKED ISSUE #265: Campaign: Facts
We need a similar field type for Facts, with Value / Source columns.  This will be used in as a "field_campaign_facts" field in the Campaign entity.

I've discussed building this out as an Entity with @desmondmorris and Fantini, where the Facts field would be an entity reference.  Although this is a clean solution on the back end, it would be tricky to implement as a select / add widget on the form itself.  For now, let's just implement as a Fact field type and create a mutli-value field on the Campaign entity.

### PR DESCRIPTION
**Pull request by aaronschachter**
Creates a Fact entity, and places entity reference fields on the campaign.

Uses ctools modal to place an "add fact" form on the campaign form.

Resolves #265

### CODE REVIEW

File: modules/dosomething/dosomething_fact/dosomething_fact.info [hunk 1]
```diff
@@ -0,0 +1,6 @@
+name = DoSomething Fact
+description = Provides a custom Fact Entity and its functionality.
+core = 7.x
+package = DoSomething
+dependencies[] = entity
```
Comment (angaither) [inline modules/dosomething/dosomething_fact/dosomething_fact.info:5]: entity reference?

File: modules/dosomething/dosomething_fact/dosomething_fact.module [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    'title callback' => t('Fact'),
+    'page callback' => 'dosomething_fact_view_entity',
+    'page arguments' => array(1, 2),
+    'access callback' => TRUE,
+  );
+  // This menu callback needs to be visible by SecurePages in order for ctools modal to work.
+  $items['admin/dosomething-fact/modal-add/%ctools_js'] = array(
+    'page callback' => 'dosomething_fact_modal_add_callback',
+    'page arguments' => array(3),
+    'access callback' => TRUE,
```
Comment (angaither) [inline modules/dosomething/dosomething_fact/dosomething_fact.module:50]: permissions?

File: modules/dosomething/dosomething_fact/dosomething_fact.info (hunk 1 above)
Comment (aaronschachter) [inline modules/dosomething/dosomething_fact/dosomething_fact.info:5]: It's been added as a dependency to dosomething_campaign, but isn't necessary for the fact.  The fact just has a few simple columns on the table, and depends on Entity to define the Fact entity.

**Comment by angaither** [comment]
Other than those two comments, looks ok. :thumbsup:

File: modules/dosomething/dosomething_fact/dosomething_fact.module (hunk 2 above)
Comment (aaronschachter) [inline modules/dosomething/dosomething_fact/dosomething_fact.module:50]: Good catch.  This will be handled by #289

**Pull request merged**